src/vos/vea/vea_api.c

```diff
@@ -402,26 +403,25 @@ process_resrvd_list(struct vea_space_info *vsi, struct vea_hint_context *hint,
 		seq_max = resrvd->vre_hint_seq;
 		off_p = resrvd->vre_blk_off + resrvd->vre_blk_cnt;
 
-		if (vfe.vfe_blk_cnt == 0) {
-			vfe.vfe_blk_off = resrvd->vre_blk_off;
-			vfe.vfe_blk_cnt = resrvd->vre_blk_cnt;
-		} else if (vfe.vfe_blk_off + vfe.vfe_blk_cnt ==
-			   resrvd->vre_blk_off) {
-			vfe.vfe_blk_cnt += resrvd->vre_blk_cnt;
-		} else if (vfe.vfe_blk_cnt != 0) {
+		if (vfe.vfe_blk_off + vfe.vfe_blk_cnt == resrvd->vre_blk_off) {
+ 			vfe.vfe_blk_cnt += resrvd->vre_blk_cnt;
+			continue;
+		}
+
+		if (vfe.vfe_blk_cnt != 0) {
 			rc = publish ? persistent_alloc(vsi, &vfe) :
 				       compound_free(vsi, &vfe, flags);
-			vfe.vfe_blk_cnt = 0;
-			if (rc)
-				goto error;
+ 			if (rc)
+ 				goto error;
+ 		}
```

Comment:
(style)  code indent should use tabs where possible

(style)  please, no space before tabs